### DISCUSSION TITLE
headless client - parameter for setting musicians name

### DISCUSSION THREAD

**Discussion by dora71** [jamulussoftware/jamulus#1047] [Ideas]
Please add a command line option for a headless client to set the own name. If it is already there, I didn't find a hint in the descriptions.

Until now, it just shows up as "No Name" with an ear as icon (that is ok), but I want to be able to change the "No Name" to make it more recognizable for the other participating musicians.

Thanks a lot.

Rainer

**Comment by gilgongo** [comment]
I'm not sure about headless, but does `--clientname` do what you need? It sets the window title and Jack client name.

**Comment by dora71** [comment]
> I'm not sure about headless, but does `--clientname` do what you need? It sets the window title and Jack client name.

@gilgongo Unfortunately not. It only does what is discribed ... setting the Jack client name and the window title, but not the "Slider Name" what the other musicians do see.

**Comment by WolfganP** [comment]
As a workaround you can edit the `jamulus.ini` file, key `<name_base64>` (tons of sites convert any string to base64 for you to customize your instance name)

**Comment by jerogee** [comment]
@dora71, if you are on a Mac or a Linux box, you could get @WolfganP's workaround to work with a single one-liner! 
```bash
$ NAME="dora"; sed -i -e "s/.*name_base64.*/ <name_base64>$(echo -n $NAME | base64)<\/name_base64>/" ~/.config/Jamulus/Jamulus.ini
```
Then you can start the client as usual, e.g.:
```bash
$ /Applications/Jamulus.app/Contents/MacOS/Jamulus --nogui --connect jamulus.jamtogether.io
```

**Comment by DominikSchaller** [comment]
You need to use the ini file for this.

**Comment by ann0see** [comment]
I'll convert this to a discussion since a solution is given (config file)
